compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -783,15 +775,17 @@ impl<'hir> Map<'hir> {
     }
 
     pub fn expect_foreign_item(&self, id: HirId) -> &'hir ForeignItem<'hir> {
-        match self.find(id) {
-            Some(Node::ForeignItem(item)) => item,
+        let did = id.expect_owner();
+        match self.tcx.hir_owner(did) {
+            Some(Owner { node: OwnerNode::ForeignItem(item) }) => item,
             _ => bug!("expected foreign item, found {}", self.node_to_string(id)),
         }
     }
 
     pub fn expect_macro_def(&self, id: HirId) -> &'hir MacroDef<'hir> {
-        match self.find(id) {
-            Some(Node::MacroDef(macro_def)) => macro_def,
+        let did = id.expect_owner();
+        match self.tcx.hir_owner(did) {
```

Comment:
```suggestion
        match self.tcx.hir_owner(id.expect_owner()) {
```
Nit: I personally have an allergy to abbreviating `def_id` to `did` and would prefer to normalize to `def_id` everywhere, but it looks like here (and in few cases above) the separate variable isn't actually necessary at all.